### ISSUE TITLE
Distribute automator

### ISSUE THREAD

**Issue by squaremo** [weaveworks/flux#207]
The automator currently works like this:

```
for each instance
  for each service marked automated in the instance
    put a "update all images for the service" job in the job queue
```

As identified in #152, this has a few scaling problems; principally that the more automators are running, the more jobs get scheduled redundantly.

**Comment by paulbellamy** [comment]
There are three immediately obvious ways of solving this, discussed below:
1) Only run a single automator per instance. Either add a cli flag and only run one globally, or shard them somehow.
  - This sucks
2) Make creating release jobs idempotent somehow
  - Maybe a deterministic ID which you can use to dedupe them, or the workers could only run the latest release job for a service/image pair.
  - Could do a `RunJobIfNotExists` which checks the existence of some other job...
3) Do the automator checking as a queued job. Each time it finishes, it re-enqueues itself for 15 seconds in the future.
  - Could be tricky to get the continuity right, though I suppose it is easy to rebuild which jobs should exist based on which services have automation enabled.
  - Generalizes nicely into a "background jobs" thing, which we could also use for checking which services have automation enabled/disabled and making sure the right jobs are scheduled...

**Comment by paulbellamy** [comment]
For the record, I reckon: 2 is easier, while 3 is a "more complete solution" (i.e. probably overengineering)

**Comment by squaremo** [comment]
> 2. Make creating release jobs idempotent somehow

Release jobs are also used when someone says `fluxctl release ...`, so we'd have to also distinguish between scheduled jobs and those; and, between jobs that had been scheduled automatically, but are due to run again.

> 3. Do the automator checking as a queued job. Each time it finishes, it re-enqueues itself for 15 seconds in the future.

Kubernetes might help us with that, but even if it doesn't, maintaining a schedule isn't too hard. I like this option because it scales horizontally (you can avoid doing redundant work -- table scans -- when you have more workers).

**Comment by paulbellamy** [comment]
Ok, 3 sounds like a winner.

Plan to do it:
1) Refactor job queue to support multiple job types
2) Add a CheckAutomatedServiceJob
3) Hook that up so it gets started, and stops itself when you enable/disable automation on a service.

**Comment by paulbellamy** [comment]
Fixed in #281
